IdentityCore/src/cache/mac/MSIDMacACLKeychainAccessor.m

```diff
@@ -49,29 +49,36 @@ - (nullable instancetype)initWithTrustedApplications:(nullable NSArray *)trusted
     self = [super init];
     if (self)
     {
-        NSArray *appList = [self trustedAppListWithCurrentApp:error];
-        
-        if (![appList count])
+        NSMutableArray *allTrustedApps = [NSMutableArray new];
+        NSArray *appList;
+        if (![trustedApplications count])
         {
-            return nil;
+            appList = [self trustedAppListWithCurrentApp:error];
+            if (![appList count])
+            {
+                return nil;
+            }
+            [allTrustedApps addObjectsFromArray:appList];
+        }
+        else
+        {
+             [allTrustedApps addObjectsFromArray:trustedApplications];
```

Comment:
Wouldn't you still want to include current app even if trusted applications list is provided? This change seems to make an assumption that trustedApplications already includes current app, but I'm not confident this is always the case. Could we instead add an additional API with a boolean parameter, e.g.

- (nullable instancetype)initWithTrustedApplications:(nullable NSArray *)trustedApplications
                                      skipTrustingCurrentApp:(BOOL)skipTrustingCurrentApp
                                         accessLabel:(nonnull NSString *)accessLabel
                                               error:(NSError * _Nullable __autoreleasing * _Nullable)error 

Current API can call into this new overload with skipTrustingCurrentApp=NO to preserve current behavior for backward compatibility.

Reply:
https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1864032 for tracking this change further. Thank you Olga.

Reply:
Sunanda, I know we spoke at length about the perf impact of Sec* APIs that access signature information, reading keychain data, etc. SecTrustedApplicationCreateFromPath may not be as expensive as those. By any chance, did you (1) establish that it in fact has a non-negligible perf impact? And (2) in case it does,  SecTrustedApplicationCreateFromPath result can be probably safely cached. Its result cannot not change over the application session lifetime. And if we call it only once, this should not be a problem anyway, right?